packages/nuka/src/hooks/use-transition.ts

```diff
@@ -0,0 +1,48 @@
+import { useEffect, useState, useRef } from 'react';
+import { D3EasingFunctions } from 'src/types';
+import * as d3Ease from 'victory-vendor/d3-ease';
```

Comment:
We have victory vendor as a development dependency, but this is production usage. It’s also worth having a discussion of whether for production victory vendor is appropriate because it brings in a lot of extra unrelated D3 packages. For frontend builds, tree shaking should take care of most of the extra stuff, but it does add extra stuff to like a node server side package

Reply:
Ah, I totally misunderstood. 

I was actually wondering, @gksander, what your thoughts are on taking my `useTransition()` hook and turning it into a lightweight animation library? And then I can create a `d3-ease` vendor package inside of that library and we can remove `victory-vendor` entirely from `nuka-carousel`.

Reply:
I updated this hook (it had since been renamed to `useTween`) to use the standard `d3-ease` package instead of the `victory-vendor` one. I ended up reducing the flexibility of the hook by changing the generic `deps` array to a couple specific variables to have the internal `useEffect` watch. It was a trade-off I made for the sake of avoiding future bugs related to stale values, etc., and unfortunately makes it less of a candidate to become its own library.

Reply:
Moving to `d3-ease` broke things due to [the same ESM issues that Victory had](https://formidable.com/blog/2022/victory-esm/). I will follow up on slack (#nuka-carousel) to discuss this.